### DISCUSSION TITLE
Is it possible to fork a sandbox programatically

### DISCUSSION THREAD

**Discussion by YarivGilad** [codesandbox/codesandbox-client#4174] [General] [answered]
Hi all, I have a particular use-case that requires an auto fork of a sandbox link submitted to my server. Is it possible? I looked at the documentation and got confused...

**Comment by garethx** [comment, marked as answer]
We don't yet have an official public API for this, but we use a POST request to https://codesandbox.io/api/v1/sandboxes/sandbox_ID/fork, for forking. You get back an ID of a forked sandbox that's anonymously owned.

**Reply by YarivGilad** [reply]
Hey @garethx thank you for responding.
I'm not sure I follow, if someone hands me a url of a sandbox of theirs and I want to programmatically fork it to my codesandbox account, is it possible and should I pass any credentials to id my account?
I tested the POST api endpoint you mentioned using Postman and got back a full json response.
Not sure which field am I looking for but what I'm aiming for is that the forked sandbox will exist in my codesandbox account... Is it possible? 
Assume the origin sandbox is public and belongs to someone else's account...

**Reply by garethx** [reply]
No, the forked sandbox is anonymously owned.

**Reply by YarivGilad** [reply]
I ended up writing a script in puppeteer to mimic manual forking. Is that cool?
I'm teaching a course and send students assignments. they submit their homework sandbox url and I need a mechanism that will make sure they are not modifying the sandbox after they have submitted the end result. hence the need to fork that url automatically...

**Reply by garethx** [reply]
Sure thing - nice solution :)

**Comment by webriq** [comment]
Related question, after forking, does it automatically start the container and we can wait for X minutes for the preview to show? It doesn't seem to start it automatically when I tried it.

**Reply by andrasbacsai** [reply]
You can programatically start a server side container with a normal http request.

For example:
`curl -H 'Accept: application/json' https://node.sse.codesandbox.io/`

The important thing is the **Accept** header. You can replace the **node** in the URL with your sandbox ID.
